src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,136 +393,140 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
-                                            current_best_class);
+    const auto End = end(out_ways);
+    // dead end, or otherwise no exitable ways
+    if (out_ways.size() == 0)
+        return 0;
 
-        const bool other_is_better_choice_by_priority =
-            best != 0 && obviousByRoadClass(in_data.road_classification,
-                                            current_best_class,
-                                            out_data.road_classification);
+    // sort out ways by what they share with the in way
+    const auto sort_by = [](const out_way &lhs, const out_way &rhs) {
+        auto left_tie = std::tie(
+            rhs.deviation_from_straight, lhs.same_classification, lhs.same_or_higher_priority);
+        auto right_tie = std::tie(
+            lhs.deviation_from_straight, rhs.same_classification, rhs.same_or_higher_priority);
+        return left_tie > right_tie;
+    };
+    std::stable_sort(begin(out_ways), End, sort_by);
+
+    // the ideal obvious turn candidate is the first way sorted to have the smallest deviation from
+    // straight and is same or higher priority class
+    const auto sameOrHigher = [&](const out_way &way) {
+        return way.same_or_higher_priority == true;
+    };
+    auto best_candidate_it = std::find_if(begin(out_ways), End, sameOrHigher);
+
+    // if no candidate is found with same or higher priority, use first non-low priority class
+    const auto notLowPriority = [&](const out_way &way) {
+        return !node_based_graph.GetEdgeData(way.road->eid)
+                    .road_classification.IsLowPriorityRoadClass();
+    };
+    if (best_candidate_it == End)
+        best_candidate_it = std::find_if(begin(out_ways), End, notLowPriority);
+
+    if (best_candidate_it != End)
+    {
+        best = best_candidate_it->index;
+        best_deviation = best_candidate_it->deviation_from_straight;
+    }
 
-        if ((!other_is_better_choice_by_priority && deviation < best_deviation) ||
-            is_better_choice_by_priority)
+    // double check if the sorted lowest deviation from straight would still be better choice
+    if (static_cast<uint32_t>(out_ways[0].index) != best)
+    {
+        const out_way first = out_ways[0];
+        const auto deviation_diff =
+            (best_deviation - first.deviation_from_straight) > FUZZY_ANGLE_DIFFERENCE;
+        const auto not_ramp_class = !node_based_graph.GetEdgeData(intersection[first.index].eid)
+                                         .road_classification.IsRampClass();
+        const auto not_link_class = !node_based_graph.GetEdgeData(intersection[first.index].eid)
+                                         .road_classification.IsLinkClass();
+        if (deviation_diff && notLowPriority(first) && not_ramp_class && not_link_class)
         {
-            best_deviation = deviation;
-            best = i;
+            best = first.index;
+            best_deviation = first.deviation_from_straight;
```

Comment:
what would this set-up end up preferring in:

```

                  .      d
a - - - b - - - - c

with the ways
`abd, motorway`
and `bc` tertiary.
```
From what I read this would end up substituting the perfectly good continuing road with a `continue slight left` and offer only a `new name` for going from `a` to `c`.

For example here, the turn to the right might actually be small enough as an angle. (Note: they also all share the same name. A nice present from the road planners in Germany)

<img width="744" alt="screen shot 2016-11-14 at 09 45 05" src="https://cloud.githubusercontent.com/assets/12932279/20258262/1e010b26-aa4f-11e6-8598-49da3a4f2150.png">